openjdk/src/test/java/org/conscrypt/PlatformTestObjectUtil.java

```diff
@@ -0,0 +1,41 @@
+package org.conscrypt;
+
+import java.security.AlgorithmConstraints;
+import java.security.AlgorithmParameters;
+import java.security.CryptoPrimitive;
+import java.security.Key;
+import java.util.Set;
+import javax.net.ssl.SNIMatcher;
+import javax.net.ssl.SNIServerName;
+
+final class PlatformTestObjectUtil {
+    static SNIMatcher newSniMatcher() {
+        return new SNIMatcher(0) {
+            @Override
+            public boolean matches(SNIServerName sniServerName) {
+                return false;
+            }
+        };
+    }
+
+    static AlgorithmConstraints newAlgorithmConstraints() {
```

Comment:
Similarly, the name here should be clear about what the resulting behavior is.

Reply:
Extracted to RestrictedAlgorithmConstraints